lighthouse-core/audits/third-party-summary.js

```diff
@@ -57,41 +76,90 @@ class ThirdPartySummary extends Audit {
    * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
    * @param {Array<LH.Artifacts.TaskNode>} mainThreadTasks
    * @param {number} cpuMultiplier
-   * @return {Map<ThirdPartyEntity, {mainThreadTime: number, transferSize: number, blockingTime: number}>}
+   * @return {{byEntity: Map<ThirdPartyEntity, Summary>, byURL: Map<string, Summary>, urls: Map<ThirdPartyEntity, string[]>}}
    */
-  static getSummaryByEntity(networkRecords, mainThreadTasks, cpuMultiplier) {
-    /** @type {Map<ThirdPartyEntity, {mainThreadTime: number, transferSize: number, blockingTime: number}>} */
-    const entities = new Map();
-    const defaultEntityStat = {mainThreadTime: 0, blockingTime: 0, transferSize: 0};
+  static getSummaries(networkRecords, mainThreadTasks, cpuMultiplier) {
+    const /** @type {Map<string, Summary>} */ byURL = new Map();
+    const /** @type {Map<ThirdPartyEntity, Summary>} */ byEntity = new Map();
+    const defaultStat = {mainThreadTime: 0, blockingTime: 0, transferSize: 0};
 
     for (const request of networkRecords) {
-      const entity = thirdPartyWeb.getEntity(request.url);
-      if (!entity) continue;
-
-      const entityStats = entities.get(entity) || {...defaultEntityStat};
-      entityStats.transferSize += request.transferSize;
-      entities.set(entity, entityStats);
+      const urlStat = byURL.get(request.url) || {...defaultStat};
+      urlStat.transferSize += request.transferSize;
+      byURL.set(request.url, urlStat);
     }
 
     const jsURLs = BootupTime.getJavaScriptURLs(networkRecords);
 
     for (const task of mainThreadTasks) {
-      const attributeableURL = BootupTime.getAttributableURLForTask(task, jsURLs);
-      const entity = thirdPartyWeb.getEntity(attributeableURL);
-      if (!entity) continue;
+      const attributableURL = BootupTime.getAttributableURLForTask(task, jsURLs);
 
-      const entityStats = entities.get(entity) || {...defaultEntityStat};
+      const urlStat = byURL.get(attributableURL) || {...defaultStat};
       const taskDuration = task.selfTime * cpuMultiplier;
       // The amount of time spent on main thread is the sum of all durations.
-      entityStats.mainThreadTime += taskDuration;
+      urlStat.mainThreadTime += taskDuration;
       // The amount of time spent *blocking* on main thread is the sum of all time longer than 50ms.
       // Note that this is not totally equivalent to the TBT definition since it fails to account for FCP,
       // but a majority of third-party work occurs after FCP and should yield largely similar numbers.
-      entityStats.blockingTime += Math.max(taskDuration - 50, 0);
-      entities.set(entity, entityStats);
+      urlStat.blockingTime += Math.max(taskDuration - 50, 0);
+      byURL.set(attributableURL, urlStat);
+    }
+
+    // Map each URL's stat to a particular third party entity.
+    const /* Map<ThirdPartyEntity, string[]> */ urls = new Map();
```

Comment:
```suggestion
    // Map each URL's stat to a particular third party entity.
    /** @type {Map<ThirdPartyEntity, string[]>} */
    const urls = new Map();
```

Reply:
Oops, force of habit from the closure compiler